middleware/tests/ledger/hsm2dongle_cmds/test_signer_heartbeat.py

```diff
@@ -0,0 +1,126 @@
+            "tweak": "aabbccddeeff",
+            "signature": HSM2DongleSignature(bytes.fromhex(self.SIG))
+        }), self.hsm2dongle.get_signer_heartbeat("1133557799aa"))
+
+        self.assert_exchange([
+            bytes.fromhex("60011133557799aa"),
+            bytes.fromhex("6002"),
+            bytes.fromhex("6003"),
+            bytes.fromhex("6004"),
+            bytes.fromhex("6005"),
+        ])
+
+    def test_error_result(self):
+        self.dongle.exchange.side_effect = [
+            b"",
+            bytes.fromhex("aabbcc" + self.SIG),
+            CommException("an-error-result", 0x6b01)
+        ]
+
+        self.assertEqual((False,), self.hsm2dongle.get_signer_heartbeat("1133557799aa"))
+
+        self.assert_exchange([
+            bytes.fromhex("60011133557799aa"),
+            bytes.fromhex("6002"),
+            bytes.fromhex("6003"),
+        ])
+
+    def test_exception(self):
+        self.dongle.exchange.side_effect = [
+            b"",
+            CommException("an-exception")
+        ]
+
+        with self.assertRaises(HSM2DongleBaseError):
+            self.hsm2dongle.get_signer_heartbeat("1133557799aa")
+
+        self.assert_exchange([
+            bytes.fromhex("60011133557799aa"),
+            bytes.fromhex("6002"),
+        ])
+
+    # def test_onboard_wipe_error(self):
+    #     self.dongle.exchange.side_effect = [bytes([0])]*(32 + 5) + [bytes([0, 1, 0])]
+
+    #     with self.assertRaises(HSM2DongleError):
+    #         self.hsm2dongle.onboard(bytes(map(lambda i: i*2, range(32))), b"1234")
+
+    #     seed_exchanges = list(map(lambda i: [0x44, i, i*2], range(32)))
+    #     pin_exchanges = [[0x41, 0, 4]] + list(
+    #         map(lambda i: [0x41, i + 1, ord(str(i + 1))], range(4)))
+    #     exchanges = seed_exchanges + pin_exchanges + [[0x07]]
+    #     timeouts = [None]*len(exchanges)
+    #     timeouts[-1] = HSM2Dongle.ONBOARDING.TIMEOUT
+    #     self.assert_exchange(exchanges, timeouts)
+
+    # def test_onboard_pin_error(self):
+    #     self.dongle.exchange.side_effect = [bytes([0])]*(32 + 3) + [
+    #         CommException("an-error")
+    #     ]
+
+    #     with self.assertRaises(HSM2DongleError):
+    #         self.hsm2dongle.onboard(bytes(map(lambda i: i*2, range(32))), b"1234")
+
+    #     seed_exchanges = list(map(lambda i: [0x44, i, i*2], range(32)))
+    #     pin_exchanges = [[0x41, 0, 4]] + list(
+    #         map(lambda i: [0x41, i + 1, ord(str(i + 1))], range(3)))
+    #     exchanges = seed_exchanges + pin_exchanges
+    #     self.assert_exchange(exchanges)
+
+    # def test_onboard_seed_error(self):
+    #     self.dongle.exchange.side_effect = [bytes([0])]*30 + [CommException("an-error")]
+
+    #     with self.assertRaises(HSM2DongleError):
+    #         self.hsm2dongle.onboard(bytes(map(lambda i: i*2, range(32))), b"1234")
+
+    #     seed_exchanges = list(map(lambda i: [0x44, i, i*2], range(31)))
+    #     self.assert_exchange(seed_exchanges)
```

Comment:
Are those commented-out tests supposed to be here?

Reply:
Good catch, thanks!